packages/eslint-plugin/src/types/rules.ts

```diff
@@ -4,9 +4,9 @@ type CodePath = Rule.CodePath;
 type CodePathSegment = Rule.CodePathSegment;
 
 interface TSRuleListener {
-  onCodePathStart?(codePath: CodePath, node: TSESTree.Node): void;
+  onCodePathStart?: (codePath: CodePath, node: any) => void;
 
-  onCodePathEnd?(codePath: CodePath, node: TSESTree.Node): void;
+  onCodePathEnd?: (codePath: CodePath, node: any) => void;
```

Comment:
Why are we maintaining this type ourselves? How come there isn't a type for `Visitor`?

Reply:
I just simplified this a bit as we don't need most of those properties, but ultimately it is because of the discussion we had. There is a vanilla visitor (they call it `Listener`, but that is for `ESTree.Node`, not `TSESTree.Node`, and `TSESTree` is a namespace from typescript eslint.